neptune-bulk-loader/index-terms.ts

```diff
@@ -0,0 +1,193 @@
+
+const fetchTermsResponseSchema = z.object({
+  results: z.object({
+    bindings: z.array(
+      z.object({
+        label: z.object({
+          value: z.string(),
+        }),
+        s: z.object({
+          value: z.string(),
+        }),
+      })
+    ),
+  }),
+});
+
+interface FetchedTerm {
+  label: string
+  uri: string
+}
+
+interface FetchTermsParams {
+  offset: number
+  sparqlUrl: string
+}
+
+const fetchTerms = async (params: FetchTermsParams): Promise<FetchedTerm[]> => {
+  const {
+    offset,
+    sparqlUrl,
+  } = params;
+
+  // TODO OBP-290 is going to make this dynamic
+  const query = `
+SELECT DISTINCT ?s ?label
+FROM <http://purl.unep.org/sdg/sdgio.owl>
+WHERE {
+  ?s a owl:Class .
+  ?s rdfs:label ?label .
+  FILTER regex(str(?s), "SDGIO_")
+}
+ORDER BY ?label
+LIMIT 200
+OFFSET ${offset}`;
+
+  const response = await got.post(
+    sparqlUrl,
+    {
+      form: { query },
+      responseType: 'json',
+      https: httpsOptions(sparqlUrl),
+      throwHttpErrors: false,
+    }
+  );
+
+  if (response.statusCode !== 200) {
+    throw new Error(`Neptune request failed with status ${response.statusCode}: ${response.body}`);
+  }
+
+  const parsedResponseBody = fetchTermsResponseSchema.parse(response.body);
+
+  const { bindings } = parsedResponseBody.results;
+
+  return bindings.map((b) => ({
+    label: b.label.value,
+    uri: b.s.value,
+  }));
+};
+
+const indexForTerm = (term: FetchedTerm, indexName: string): unknown => ({
+  index: {
+    _index: indexName,
+    _type: 'doc',
+    _id: term.uri,
+  },
+});
+
+const queryForTerm = (
+  term: FetchedTerm,
+  terminologyTitle: string,
+  ontologyGraph: string
+): unknown => ({
+  query: {
+    multi_match: {
+      query: term.label,
+      type: 'phrase',
+      fields: ['contents', 'title'],
+    },
+  },
+  source_terminology: terminologyTitle,
+  ontologyGraph,
+});
+
+interface BulkIndexTermsParams {
+  elasticsearchClient: Got
+  terms: FetchedTerm[]
+  indexName: string
+  terminologyTitle: string
+  ontologyGraph: string
+}
+
+const bulkIndexTerms = async (params: BulkIndexTermsParams): Promise<void> => {
+  const {
+    elasticsearchClient,
+    terms,
+    indexName,
+    terminologyTitle,
+    ontologyGraph,
+  } = params;
+
+  const esDoc = _(terms)
+    .map((term) => [
+      indexForTerm(term, indexName),
+      queryForTerm(term, terminologyTitle, ontologyGraph),
+    ])
+    .flatten()
+    .map((x) => JSON.stringify(x))
+    .join('\n');
+
+  const body = `${esDoc}\n`;
+
+  const response = await elasticsearchClient.post(
+    `${indexName}/doc/_bulk`,
+    {
+      headers: { 'Content-Type': 'application/json' },
+      body,
+      throwHttpErrors: false,
+    }
+  );
+
+  if (response.statusCode !== 200) {
+    throw new Error(`Neptune request failed with status ${response.statusCode}: ${response.body}`);
+  }
+};
+
+interface CreateTermIndexParams {
+  elasticsearchUrl: string
+  ontologyNameSpace: string
+  ontologyGraph: string
```

Comment:
This is a small thing but in this file it's called `ontologyGraph` but it comes from `ontologyGraphUrl` - which I think in Neptune is actually `namedGraphUri`). It doesn't really matter what it's called just would like it to be consistent - - unless there is a reason you've made them different.

Reply:
Fixed in 33fae9e